app/models/king.rb

```diff
@@ -1,7 +1,10 @@
 class King < Piece
 
   def valid_move?(x_destination, y_destination)
-    (cell_x - x_destination).abs <= 1 && (cell_y - y_destination).abs <= 1 && is_obstructed?([x_destination, y_destination])
+    return false unless (cell_x - x_destination).abs <= 1 && (cell_y - y_destination).abs <= 1  
```

Comment:
I don't think you need "return false unless" on this line. Ruby will return the last line as default.